app/furniture/marketplace/square_order.rb

```diff
@@ -0,0 +1,167 @@
+# and automatically notify sellers of new orders.
+#
+# NOTE: the simple `Markeplace > SquareOrder` namespacing was chosen
+# to avoid collision with the `Square` namespace already included by Square's gem.
+#
+# TODOS/considerations:
+# 1. Implement validation of `order` argument on initialization
+#   Eg: if order.is_not_valid?; exit; end
+# 2. Implement state tracking
+#   Eg: READY | ORDER_SENT | etc...
+class Marketplace
+  class SquareOrder
+    def initialize(order)
+      @order = order
+      @marketplace = order.marketplace
+      @shopper = order.shopper
+      @space = order.marketplace.space
+      @ordered_products = @order.ordered_products
+      @square_location_id = @marketplace.settings["square_location_id"]
+      @square_order_id = nil
+      @square_payment_id = nil
+      # @state = "READY"
+    end
+
+    # Sends an order to Square. Will create both an order and payment in the
+    # receiving account which are requirements for the order to show up in the
+    # Seller Dashboard.
+    def send_order
+      Rails.logger.info("Start creating Square order")
+      square_create_order_response = create_square_order
+      Rails.logger.info("Finished creating Square order")
+
+      if square_create_order_response
+        @square_order_id = square_create_order_response.body.order[:id]
+        Rails.logger.info("Start creating Square order payment")
+        square_create_payment_response = create_square_order_payment
+        Rails.logger.info("Finished creating Square order payment")
+        @square_payment_id = square_create_payment_response.body.payment[:id]
+        # @state = "ORDER_SENT"
+
+        # This data is intended for use in debugging, etc... until we further
+        # the Square integration productize
+        {
+          order_id: @square_order_id,
+          payment_id: @square_payment_id
+        }
+      else
+        # TODO: Noop for now
+      end
+    end
+
+    # Square sets max of 43 chars so this assumes an order id remains a 36-char
+    # uuid
+    private def square_idemp_key
+      "#{@order.id}_#{8.times.map { rand(10) }.join}"
+    end
+
+    private def create_square_order
+      square_create_order_body = prepare_square_create_order_body(@marketplace)
+      @square_client.orders.create_order(body: square_create_order_body)
+    end
+
+    # NOTE: Square requires that orders must have be in a state of complete
+    # payment to display in the Seller Dashboard
+    private def create_square_order_payment
+      square_create_payment_body = prepare_square_create_order_payment_body
+
+      @square_client.payments.create_payment(body: square_create_payment_body)
+    end
+
+    # NOTE: Square requires that orders must include fulfillments to display
+    # in the Seller Dashboard
+    # See: https://developer.squareup.com/docs/orders-api/create-orders
+    private def prepare_square_create_order_body
+      location_id = @marketplace.settings["square_location_id"]
```

Comment:
I believe this is available on the `marketplace` as `marketplace.square_location_Id`

Reply:
Yeah, not sure why I did that.